File: atcoder/abc138C_alchemist.cpp
// TsukiZombina
// https://atcoder.jp/contests/abc138/tasks/abc138_c
// sorting, greedy

#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N;
    float number = 0, mix;
    std::cin >> N;
    std::vector<int> numbers(N);
    for(auto& x: numbers)
    {
        std::cin >> x;
    }

    std::sort(numbers.begin(), numbers.end());
     
    mix = numbers[0];
    for (const auto& number: numbers) {
        mix = (mix + number)/2;
    }
    std::cout << mix;
    return 0;
}

File: atcoder/abc138C_alchemist.ml
(* vicfred
 * https://atcoder.jp/contests/abc138/tasks/abc138_c
 * sorting, greedy
 * *)

let main =
    Scanf.scanf "%d\n" (fun n ->
        Array.init n (fun _ -> Scanf.scanf " %f" (fun f -> f))
        |> Array.to_list
        |> List.sort compare
        |> (fun arr -> List.fold_left (fun x y -> (x +. y) /. 2.0) (List.hd arr) arr)
        |> Printf.printf "%.9f\n"
)
